cinspect/model_evaluation.py

```diff
@@ -98,31 +96,25 @@ def bootstrap_model(
     # Run various checks and prepare the evaluators
     indices = np.arange(len(X))
     random_state = check_random_state(random_state)
-    groups = _check_group_estimator(estimator, use_group_cv)
-    for ev in evaluators:
-        ev.prepare(estimator, X, y, random_state=random_state)
 
-    LOG.info("Bootstrapping ...")
-
-    # Bootstrapping loop
-    for i in range(replications):
-        LOG.info(f"Bootstrap round {i + 1}")
-        start = time.time()
-
-        Xb, yb, indb = resample(X, y, indices, random_state=random_state)
-        estimator.fit(Xb, yb, groups=indb) if groups else estimator.fit(Xb, yb)
-        for ev in evaluators:
-            ev.evaluate(estimator, Xb, yb)
-
-        end = time.time()
-        LOG.info(f"... iteration time {end - start:.2f}s")
-
-    LOG.info("Bootstrapping done.")
-
-    for ev in evaluators:
-        ev.aggregate()
-
-    return evaluators
+    bootstrap_split_generator = (
+        # identical train, test sets TODO: should we be evaluating on the whole data set?
+        (split1 := resample(indices, random_state=random_state), split1)
```

Comment:
Yes I think that TODO was a hold-over from earlier thinking; I'll remove it. It's now pretty easy to make your own bootstrap-/cross-val-like functions, so that comment is redundant. I only recently found out about the 'walrus'... I was a little nervous about using a new feature (for readability's sake), but hopefully the meaning is apparent from context here